tools/pre-commit/README.md

```diff
@@ -0,0 +1,141 @@
+# `pre-commit`
+
+`pre-commit` is a useful tool used by developers to ensure code quality and
+consistency before committing changes to Git. It automates the process of
+running various checks and tests on your codebase, helping catch potential
+issues early on and maintaining a clean and reliable codebase.
+
+More concretely, the `pre-commit` tool is a framework that is capable of
+managing and running various multi-language tools, called _hooks_, whenever a
+specific [Git hook] runs.
+
+[Git hook]: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
+
+A `.pre-commit-commit.yaml` file, located in the project root dir, defines
+exactly which hooks should run, with which arguments, and when.
+
+See our `.pre-commit-commit.yaml` file in the [irnas-zephyr-template]
+repository.
+
+[irnas-zephyr-template]: https://github.com/IRNAS/irnas-zephyr-template
+
+## Setup
+
+To install `pre-commit` run:
+
+```bash
+pip install pre-commit
+```
+
+## Usage
+
+### Repository setup
+
+To make `pre-commit` run before every `git commit` you need to install it into
+the repository:
+
+```bash
+pre-commit install
+```
+
+Every time you clone a project that uses `pre-commit` you should run
+`pre-commit install` before doing anything else.
+
+### Manual runs
+
+To manually run `pre-commit` on all files in the repository you can run:
+
+```bash
+pre-commit run --all-files
+```
+
+Without the `--all-files` flag `pre-commit` runs only on currently staged files.
+
+To run only a specific hook you can run:
+
+```bash
+pre-commit run --all-files <hook-id>
+```
+
+For example:
+
+```bash
+pre-commit run --all-files clang-format
+```
+
+### Testing commit messages
+
+To test if your commit message passes the rules imposed by the `committed` tool:
+
+```bash
+pre-commit run --hook-stage commit-msg committed --commit-msg-filename <file_with_commit_message>
+```
```

Comment:
I would move this to the end of the file into a _Testing and debugging_ section

Reply:
Moved.